src/sentry/utils/snowflake.py

```diff
@@ -28,7 +29,9 @@ def save_with_snowflake_id(self, snowflake_redis_key, save_callback):
                 return
             except IntegrityError:
                 self.id = None
-        raise MaxSnowflakeRetryError
+
+        capture_exception(MaxSnowflakeRetryError())
+        raise MaxSnowflakeRetryError()
```

Comment:
What's the intent here? This looks likely to cause the error to be double-logged, since the second exception object could easily land in `capture_exception` again. (And even if that's not the case, it requires this function to know a little too much about what will happen to the exception after raising.)